Class 2.ipynb

```diff
@@ -0,0 +1,721 @@
+       "       'Found Property', 'Bridge Condition', 'Bike Rack Condition',\n",
+       "       'BEST/Site Safety', 'EAP Inspection - F59', 'Vending',\n",
+       "       'Public Assembly', 'Public Payphone Complaint', 'Unleashed Dog',\n",
+       "       'Curb Condition', 'Cranes and Derricks', 'DCA Literature Request',\n",
+       "       'City Vehicle Placard Complaint',\n",
+       "       'Legal Services Provider Complaint', 'Posting Advertisement',\n",
+       "       'PAINT - PLASTER', 'NONCONST', 'APPLIANCE', 'CONSTRUCTION',\n",
+       "       'Recycling Enforcement', 'Special Natural Area District (SNAD)',\n",
+       "       'DPR Internal', 'Opinion for the Mayor', 'Mold', 'Snow',\n",
+       "       'Complaint', 'Urinating in Public', 'Senior Center Complaint',\n",
+       "       'Animal Facility - No Permit', 'Unsanitary Animal Facility',\n",
+       "       'Parking Card', 'Fire Alarm - Reinspection',\n",
+       "       'Request for Information', 'Miscellaneous Categories',\n",
+       "       'Unsanitary Pigeon Condition', 'Plant', 'Fire Alarm - Replacement',\n",
+       "       'DPR Literature Request', 'Invitation', 'Scaffold Safety',\n",
+       "       'Transportation Provider Complaint', 'Ferry Complaint',\n",
+       "       'STRUCTURAL', 'Bus Stop Shelter Placement', 'Compliment',\n",
+       "       'Misc. Comments', 'Disorderly Youth', 'OEM Literature Request',\n",
+       "       'Fire Alarm - New System', 'Fire Alarm - Modification',\n",
+       "       'Stalled Sites', 'Public Toilet', 'Tattooing',\n",
+       "       'Illegal Animal Sold', 'Illegal Animal Kept as Pet',\n",
+       "       'DFTA Literature Request', 'Adopt-A-Basket',\n",
+       "       'DHS Income Savings Requirement', 'Poison Ivy', 'Tunnel Condition',\n",
+       "       'Drinking Water', 'Municipal Parking Facility', 'Trans Fat',\n",
+       "       'Illegal Fireworks', 'Window Guard', 'Open Flame Permit', 'DWD',\n",
+       "       'Highway Sign - Damaged', 'Ferry Permit', 'X-Ray Machine/Equipment'], dtype=object)"
+      ]
+     },
+     "execution_count": 22,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "cleaned.complaint_type.unique()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 36,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "complaint_type\n",
+       "HEATING                   14200\n",
+       "GENERAL CONSTRUCTION       7471\n",
+       "Street Light Condition     7117\n",
+       "DOF Literature Request     5797\n",
+       "PLUMBING                   5373\n",
+       "dtype: int64"
+      ]
+     },
+     "execution_count": 36,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Let's figure out what the top complaints are\n",
+    "cleaned.groupby('complaint_type').size().sort_values(ascending=False).head()\n"
```

Comment:
Maybe use `.value_counts` here instead for simplicity? Then introduce the aggregator later and show that these are equivalent